mountpoint-s3-client/src/imds_crt_client.rs

```diff
@@ -56,32 +50,30 @@ impl ImdsCrtClient {
     }
 
     /// Query the type of the EC2 instance this code is executed on.
-    pub fn make_instance_type_query(&self) -> Result<ImdsQueryRequest, ImdsQueryRequestError> {
+    pub async fn get_instance_type(&self) -> Result<String, ImdsQueryRequestError> {
```

Comment:
Not for this PR, but maybe we should switch to just grabbing the [instance identity document](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html) since we can do it in one query instead of two.

Reply:
Tempted about this: it would also "solve" the issue with spurious replies below.

Reply:
I've switched to getting the instance identity document and parsing it. This saves us a query and mitigates the issue with spurious responses, since they will fail to parse correctly. The drawback is some added complexity in `main.rs` where we create and pass around a lazy `InstanceInfo` to query the Imds once and only if required. We may eventually want to refactor the logic around that, but I would say not in this PR.